bundles/org.openhab.binding.pjlinkdevice/src/main/java/org/openhab/binding/pjlinkdevice/internal/PJLinkDeviceHandler.java

```diff
@@ -0,0 +1,313 @@
+        } catch (AuthenticationException e) {
+            this.handleAuthenticationException(e);
+        }
+    }
+
+    private void setupRefreshInterval() {
+        clearRefreshInterval();
+        boolean atLeastOneChannelToBeRefreshed = PJLinkDeviceHandler.this.config.refreshPower
+                || PJLinkDeviceHandler.this.config.refreshMute || PJLinkDeviceHandler.this.config.refreshInputChannel;
+        if (config.refresh > 0 && atLeastOneChannelToBeRefreshed) {
+            refreshJob = scheduler.scheduleWithFixedDelay(this::refresh, 0, config.refresh, TimeUnit.SECONDS);
+        }
+    }
+
+    private void updateDeviceProperties(PJLinkDevice device) throws IOException, AuthenticationException {
+        updateProperty(PJLinkDeviceBindingConstants.PARAMETER_AUTHENTICATION_REQUIRED,
+                device.getAuthenticationRequired().toString());
+
+        try {
+            updateProperty(PJLinkDeviceBindingConstants.PARAMETER_NAME, device.getName());
+        } catch (ResponseException e) {
+            // okay, cannot retrieve model information
+            updateProperty(PJLinkDeviceBindingConstants.PARAMETER_NAME, e.toString());
+        }
+        try {
+            updateProperty(PJLinkDeviceBindingConstants.PARAMETER_MANUFACTURER, device.getManufacturer());
+        } catch (ResponseException e) {
+            // okay, cannot retrieve model information
+            updateProperty(PJLinkDeviceBindingConstants.PARAMETER_MANUFACTURER, e.toString());
+        }
+        try {
+            updateProperty(PJLinkDeviceBindingConstants.PARAMETER_MODEL, device.getModel());
+        } catch (ResponseException e) {
+            // okay, cannot retrieve model information
+            updateProperty(PJLinkDeviceBindingConstants.PARAMETER_MODEL, e.toString());
+        }
+        try {
+            updateProperty(PJLinkDeviceBindingConstants.PARAMETER_COMBINED_ID, device.getFullDescription());
+        } catch (ResponseException e) {
+            // okay, cannot retrieve model information
+            updateProperty(PJLinkDeviceBindingConstants.PARAMETER_COMBINED_ID, e.toString());
+        }
+        try {
+            updateProperty(PJLinkDeviceBindingConstants.PARAMETER_CLASS, device.getPJLinkClass());
+        } catch (ResponseException e) {
+            // okay, cannot retrieve class information
+            updateProperty(PJLinkDeviceBindingConstants.PARAMETER_CLASS, e.toString());
+        }
+        try {
+            Map<ErrorStatusDevicePart, ErrorStatusQueryResponseState> errorStatus = device.getErrorStatus();
+            for (Map.Entry<ErrorStatusDevicePart, ErrorStatusQueryResponseState> entry : errorStatus.entrySet()) {
+                String key = entry.getKey().getCamelCaseText();
+                String value = entry.getValue().getText();
+                updateProperty(PJLinkDeviceBindingConstants.PARAMETER_ERROR_STATUS + key, value);
```

Comment:
I don't see real interest to provide that as property for the end user. This is more technical information.

Reply:
I am an end user and I'm interested in this technical information :-)